vltt/assignment1/Q1.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <iomanip>
+#include <string>
+#include <unordered_set>
+#include <unordered_map>
+
+//is_case_sensitive == 1 -  sensitive, is_case_sensitive == 0 - insensitive
+
+
+void split_by_word(std::string &sentence, std::unordered_map<std::string, int> &word_counts, bool is_case_sensitive) {
+    std::string cur;
+    for (int i = 0; i <= sentence.length(); i++) {
+        if (i == sentence.length() || !isalpha(sentence[i]) && !isdigit(sentence[i])) {
+            if (!cur.empty()) {
+                sort(cur.begin(), cur.end());
+                word_counts[cur]++;
+                cur = "";
+            }
+            if (i == sentence.length()) return;
+        }
+        else if (is_case_sensitive) cur += sentence[i];
+        else cur += tolower(sentence[i]);
+    }
+}
+
+bool is_anagram(std::string &sentence_a, std::string &sentence_b, bool is_case_sensitive) {
```

Comment:
Should be:

`bool is_anagram(const std::string &sentence_a, const std::string &sentence_b, bool is_case_sensitive)`